internal/blocktx/processor.go

```diff
@@ -481,38 +401,69 @@ func (ph *Processor) processBlock(msg *p2p.BlockMessage) {
 	timeStart := time.Now()
 
 	blockHash := msg.Header.BlockHash()
-
 	previousBlockHash := msg.Header.PrevBlock
-
 	merkleRoot := msg.Header.MerkleRoot
 
-	blockId, err := ph.insertBlock(ctx, &blockHash, &merkleRoot, &previousBlockHash, msg.Height)
+	prevBlock, err := p.getPrevBlock(ctx, &previousBlockHash, &blockHash)
+	if err != nil {
+		p.logger.Error("unable to get previous block from db", slog.String("hash", blockHash.String()), slog.Uint64("height", msg.Height), slog.String("prevHash", previousBlockHash.String()), slog.String("err", err.Error()))
+		return
+	}
+
+	incomingBlock := createBlock(msg, prevBlock)
+
+	competing, err := p.competingChainsExist(ctx, incomingBlock)
+	if err != nil {
+		p.logger.Error("unable to check for competing chains", slog.String("hash", blockHash.String()), slog.Uint64("height", msg.Height), slog.String("err", err.Error()))
+		return
+	}
+
+	if competing {
+		p.logger.Info("Competing blocks found", slog.String("incoming block hash", blockHash.String()), slog.Uint64("height", incomingBlock.Height))
+
+		greatestChainwork, err := p.hasGreatestChainwork(ctx, incomingBlock)
```

Comment:
nit: It would be more visible that `greatestChainwork` is boolean if it were called eg.e `hasGreatestChainwork`

Reply:
Right, changed